src/llm-service.test.ts

```diff
@@ -0,0 +1,476 @@
+import { describe, test, expect, mock, beforeEach } from "bun:test";
```

Comment:
I've tried to make few tests here, but kinda hard without fully understanding of the service